.ci/load/scripts/param_gen/gen_params.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python
+
+# Licensed to Elasticsearch B.V. under one or more contributor
+# license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright
+# ownership. Elasticsearch B.V. licenses this file to you under
+# the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+Description:
+
+This is an ad-hoc script used to generate the drop-down selections
+for the Jenkinsfile to allow users to test against various combinations
+of JDKs available in the Elastic JVM catalog which is hosted at:
+https://jvm-catalog.elastic.co.
+
+It dymamically generates a list of available JDKs and of APM Java Agent
+releases. These are returns as a snippet of code which can then be pasted
+directly into a Jenkinsfile. Below is an example of the kind of output
+produced by this script:
+
+    parameters {
+    string(name: 'agent_version', defaultValue: 'v1.9.0', description: 'Version of agent. Should correspond to tag, e.g. `v1.9.0`.')
+    string(name: 'jvm_version', defaultValue: '9.0.4', description: 'Version of JVM.')
+    string(name: 'concurrent_requests', defaultValue: '100', description: 'The number of concurrent requests to test with.')
+    string(name: 'duration', defaultValue: '10', description: 'Test duration in minutes. Max: 280')
+    string(name: 'num_of_runs', defaultValue: '1', description: 'Number of test runs to execute.')
+    }
+
+Author: Mike Place <mike.place@elastic.co>
+
+Maintainers: Observability Developer Productivity
+<observability-robots@elastic.co>
+"""
+
+import requests
+import argparse
+import github
+from packaging.version import parse as parse_version
+
+CATALOG_URL = 'https://jvm-catalog.elastic.co'
+SUPPORTED_JDKS = ['oracle', 'openjdk', 'adoptopenjdk', 'amazon', 'jdk', 'zulu']
+MIN_JDK_VERSION = '11'
```

Comment:
We'll also need to test against Java 7 and 8. Does Spring PetClinic event support such old versions?

Reply:
Good question. I might need a little help from your team to make this determination. I see Java 8 referenced here, though: https://github.com/spring-projects/spring-petclinic#prerequisites

Reply:
@felixbarny Also, with this, we are allowed only 250 entries in a drop-down list. If I set this down to 7 as the minimum, we're just over the limit. I could potentially refactor this to try and work around this but before I do, I should ask if there are any of these flavors you clearly won't be needing?

Reply:
Does removing `'amazon'` from `SUPPORTED_JDKS` bring us under the limit? If not, also exclude `'jdk'`.

Reply:
@felixbarny Hrm. Sadly, removing both `amazon` and `jdk` only gets us to 266. If we just took the most recent patch release of any minor would that be sufficient for you or do you need that level of granularity?

Reply:
Unfortunately, we do need that level of granularity so that we can try to reproduce specific issues such as JVM crashes. It also helps us to bisect the JVM update version where a particular JVM bug has been fixed that causes issues in combination with the APM agent. 
Another possibility would be to just list the Zulu builds, as they seem to offer the most exhaustive list of update versions, and to add a plain-text input for things other than Zulu.
Or we could add different dropdowns for each vendor. By default, no JDK would be selected. If more than one checkbox has a non-default value, or if all have a non-default value we can consider that an input error.

Reply:
Sounds good. Let me play with some options and see what I can come up with.

Reply:
There were some duplicates in the list that I removed. We're good now. 👍